### DISCUSSION TITLE
Reverse Toggle All Rows Selected checkbox behavior for v7

### DISCUSSION THREAD

**Discussion by wesokuhara** [TanStack/table#3244] [Q&A] [answered]
Hello `react-table` community,

I am using the `useRowSelect` plugin with React Table v7.  By default, when some (not all) rows are selected and the "Toggle All" checkbox is in the indeterminate state (represented by the "-"), all rows will be selected when the checkbox is clicked.  I am wondering if it is possible to have the opposite behavior, i.e. when the "Toggle All" checkbox is clicked in its indeterminate state, it de-selects all rows.  Is this achievable?  

![Screen Shot 2021-04-27 at 11 23 30 AM](https://user-images.githubusercontent.com/7199654/116293473-cbe37780-a74b-11eb-93a4-464f928f6b47.png)

Thanks for building great OSS.

**Comment by rossthedevigner** [comment]
@wesokuhara Have you been able to make this happen?

**Reply by wesokuhara** [reply]
I haven't spent much time on this unfortunately.  But I glanced at the code and in order for this to happen, some modification would need to be made to the reducer for the `useRowSelect` hook https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useRowSelect.js#L110

**Reply by rossthedevigner** [reply]
Appreciate you responding, and I agree with your theory.

**Reply by wesokuhara** [reply]
Hey @rossthedevigner, without making any changes to the library code, I was able to achieve this thanks to the configurability that `react-table` offers.  Basically, we can override the default props passed to the select-all checkbox, specifically the `onChange` prop, using `getToggleAllRowsSelectedProps()`.
```ts
const customHook = (existingColumns: ColumnInstance[], { instance }: MetaBase<{}>) => {
  if (instance.disableRowSelection) return existingColumns;

  const onToggleAllRowsChange = () => {
    const noRowsSelected = Object.keys(instance.state.selectedRowIds).length === 0;
    instance.toggleAllRowsSelected(noRowsSelected);
  };

  return [
    {
      Header: ({ getToggleAllRowsSelectedProps }: HeaderProps<any>) => {
        return (
          <TableRowCheckbox
            {...getToggleAllRowsSelectedProps({ onChange: onToggleAllRowsChange })}
          />
        );
      },
      // other props
    },
    ...existingColumns,
  ];
};
```

**Comment by wesokuhara** [comment, marked as answer]
Solved via https://github.com/tannerlinsley/react-table/discussions/3244#discussioncomment-900312
